src/descriptor/mod.rs

```diff
@@ -215,46 +222,59 @@ impl<K: InnerXKey> XKeyUtils for DescriptorXKey<K> {
         }
     }
 
-    fn root_fingerprint(&self) -> Fingerprint {
+    fn root_fingerprint(&self, secp: &Secp256k1<All>) -> Fingerprint {
         match self.origin {
             Some((fingerprint, _)) => fingerprint,
-            None => self.xkey.xkey_fingerprint(),
+            None => self.xkey.xkey_fingerprint(secp),
         }
     }
 }
 
 pub(crate) trait DescriptorMeta: Sized {
     fn is_witness(&self) -> bool;
-    fn get_hd_keypaths(&self, index: u32) -> Result<HDKeyPaths, Error>;
+    fn get_hd_keypaths(&self, index: u32, secp: &Secp256k1<All>) -> Result<HDKeyPaths, Error>;
```

Comment:
It might make sense to define a type for this : `pub type SecpCtx = Secp256k1<All>;`